pkg/distributor/distributorerror/errors.go

```diff
@@ -0,0 +1,150 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package distributorerror
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+)
+
+const (
+	// 529 is non-standard status code used by some services to signal that "The service is overloaded".
+	StatusServiceOverloaded = 529
+)
+
+// ReplicasNotMatch is an error stating that replicas do not match.
+type ReplicasNotMatch struct {
+	replica, elected, format string
+}
+
+// ReplicasNotMatchErrorf formats and returns a ReplicasNotMatch error.
+// The error message is formatted according to the given format specifier, replica and elected.
+// In order to print the limit and the burst, the format specifier must contain two %s verbs.
+func ReplicasNotMatchErrorf(format, replica, elected string) ReplicasNotMatch {
+	return ReplicasNotMatch{
+		replica: replica,
+		elected: elected,
+		format:  format,
+	}
+}
+
+func (e ReplicasNotMatch) Error() string {
+	return fmt.Sprintf(e.format, e.replica, e.elected)
+}
+
+// TooManyClusters is an error stating that there are too many HA clusters.
+type TooManyClusters struct {
+	limit  int
+	format string
```

Comment:
Nit, and probably will be refactored, but it _should_ be:
```suggestion
	format string
	limit  int
```
Because that's the order in which the params are defined in the constructor of the error and passes to fmt.Sprintf.